src/traces/scattergl/index.js

```diff
@@ -417,6 +432,24 @@ function plot(gd, subplot, cdata) {
         }
         if(scene.line2d) {
             scene.line2d.update(scene.lineOptions);
+            scene.lineOptions = scene.lineOptions.map(function(lineOptions) {
+                if(lineOptions && lineOptions.positions) {
```

Comment:
Can we combine this new logic here with one of the loops in

https://github.com/plotly/plotly.js/blob/master/src/traces/scattergl/convert.js#L387-L468

to alleviate potential performance loss.

Reply:
Not in the current state . You have 3 loops ... 2 of this loops currently can not be trigger because line.shape is not managed for scattergl . And the last loop is only for one case , connectgaps .

Reply:
Good eye, we can't _combine_ this loop with the existing loops in convert.js. 

That said, as your new loop over positions doesn't depend on viewport attributes, would you mind moving it over to [convert.js](https://github.com/plotly/plotly.js/blob/master/src/traces/scattergl/convert.js#L387-L468)? We try to keep computations in `ScatterGl.plot` to a minimum. Things that depend on the viewport (e.g. pixel positions) must be in `plot` as the viewport isn't known during `convert`, otherwise the heavy-lifting (which scale as `Math.min(x.length, y.length)` belongs in `convert`.

Sorry for the confusion.

Reply:
a example https://codepen.io/erwanmas/pen/qMoVWW

Reply:
> I can not push the fill to zero logic here because , this part is only used when you have lines

Hmm. I'm not sure I understand. 

You cannot push the fill to logic here [in convert.js, you mean?] because this part [the part you added in `ScatterGl.plot`] is only used when you have lines?

But this block

https://github.com/plotly/plotly.js/blob/master/src/traces/scattergl/convert.js#L387-L468

is also only used when _you have lines_.

------------



> i added some basics shape for line ( vhv/hvh) , and i activated the options for hv/vh .

Nice. Thanks! :tada:

Reply:
The part i added in ScatterGl.plot   is still used when you need to do a fill without lines ., but this block 
https://github.com/plotly/plotly.js/blob/master/src/traces/scattergl/convert.js#L387-L468  is only used whn you need to draw a line

Reply:
OK true. But there's must _still_ be another way to move this block into convert.js.

Of course, you don't have to do this (thanks again for this PR!), but at some point someone will to improve performance.